### PR TITLE
CI matrix change: debian+centos+sles

### PR DESCRIPTION
**Pull request by aik-jahoda**
Changes in CI matrix (updated):

| OS | What to do | Notes
| - | - | -
| Debian 11 © | Drop PR (as we have PR on Debian 10), keep rolling
| Centos.7 © | Add rolling 
| Centos.8 © | Drop PR (in favor of 7), keep rolling
| SLES 12 | Drop PR, Add rolling
| Fedora 34 | Drop PR, keep rolling

part of #57947

### CODE REVIEW

**Review by safern — APPROVED** [review]
Changes look good.

cc: @danmoseley

File: eng/pipelines/libraries/helix-queues-setup.yml [hunk 1]
```diff
@@ -67,21 +65,18 @@ jobs:
             - RedHat.7.Amd64.Open
             - Ubuntu.1604.Amd64.Open
             - Ubuntu.1804.Amd64.Open
-            - SLES.12.Amd64.Open
             - SLES.15.Amd64.Open
```
Comment (jkotas) [inline eng/pipelines/libraries/helix-queues-setup.yml:68]: Should `isFullMatrix==true` be a strict super set of `isFullMatrix==false`?

File: eng/pipelines/libraries/helix-queues-setup.yml [hunk 2]
```diff
@@ -67,21 +65,18 @@ jobs:
...
-            - (Debian.10.Amd64.Open)Ubuntu.1804.Amd64.Open@mcr.microsoft.com/dotnet-buildtools/prereqs:debian-10-helix-amd64-20210304164434-56c6673
             - (Debian.11.Amd64.Open)Ubuntu.1804.Amd64.Open@mcr.microsoft.com/dotnet-buildtools/prereqs:debian-11-helix-amd64-20210304164428-5a7c380
             - (Mariner.1.0.Amd64.Open)ubuntu.1604.amd64.open@mcr.microsoft.com/dotnet-buildtools/prereqs:cbl-mariner-1.0-helix-20210528192219-92bf620
           - ${{ if eq(parameters.jobParameters.isFullMatrix, false) }}:
             - (Centos.8.Amd64.Open)Ubuntu.1604.Amd64.Open@mcr.microsoft.com/dotnet-buildtools/prereqs:centos-8-helix-20201229003624-c1bf759
             - RedHat.7.Amd64.Open
             - (Debian.10.Amd64.Open)Ubuntu.1804.Amd64.Open@mcr.microsoft.com/dotnet-buildtools/prereqs:debian-10-helix-amd64-20210304164434-56c6673
-            - (Debian.11.Amd64.Open)Ubuntu.1804.Amd64.Open@mcr.microsoft.com/dotnet-buildtools/prereqs:debian-11-helix-amd64-20210304164428-5a7c380
             - Ubuntu.1604.Amd64.Open
             - Ubuntu.1804.Amd64.Open
```
Comment (jkotas) [inline eng/pipelines/libraries/helix-queues-setup.yml:78]: I would drop Ubuntu16, SLES and Fedora from the `isFullMatrix==false` runs.

RedHat+CentOS+Debian+Ubuntu should be more than good enough coverage for the various Linux flavors.

File: eng/pipelines/libraries/helix-queues-setup.yml (hunk 1 above)
Comment (safern) [reply eng/pipelines/libraries/helix-queues-setup.yml:68]: Yes, I've asked that in other PRs. That way we can track pass rate.

Comment (aik-jahoda) [reply eng/pipelines/libraries/helix-queues-setup.yml:68]: @jkotas This is game changer. If I understand correctly, we are not able run test only for PR as Rolling tests will run all PR tests, is that correct?

Comment (jkotas) [reply eng/pipelines/libraries/helix-queues-setup.yml:68]: Yes, I think it makes sense for PRs to run subset and for rolling to run everything (including the subset that PRs run).

I believe that it is the case today for the most part, there are just a few outliers.

Comment (aik-jahoda) [reply eng/pipelines/libraries/helix-queues-setup.yml:68]: @ViktorHofer, If I the a  Debian 10 © as an example: We want this update: `Drop rolling (keep PR)`, was it meant to remove it only from isFullMatrix==true and leave it in isFullMatrix==false ?

Comment (ViktorHofer) [reply eng/pipelines/libraries/helix-queues-setup.yml:68]: @aik-jahoda based on the discussion above, we don't want to drop rolling anywhere except for EOL distros. So "Drop rolling (keep PR)" should probably be ignored.

**Comment by aik-jahoda** [comment]
@jkotas , can you please review the table below to confirm we are on the same page?

| OS | What to do | Notes
| - | - | -
| Debian 11 © | Drop PR (as we have PR on Debian 10), keep rolling
| Centos.7 © | Add rolling 
| Centos.8 © | Drop PR (in favor of 7), keep rolling
| SLES 12 | Drop PR, Add rolling
| Fedora 34 | Drop PR, keep rolling
| Ubuntu 16 | Already removed in main as it is EOL

**Comment by jkotas** [comment]
@aik-jahoda Yes, this looks good. Thank you!

**Review by jkotas — APPROVED** [review]
Thanks

**Comment by aik-jahoda** [comment]
Fails on RETHEL 8 which is out of scope of this change.

**Pull request merged**